godot-core/src/registry/class.rs

```diff
@@ -559,6 +588,23 @@ fn register_class_raw(mut info: ClassRegistrationInfo) {
         info.godot_params.get_virtual_func = info.user_virtual_fn.or(info.default_virtual_fn);
     }
 
+    // If no class-level icon was set, use the extension-level icon as fallback (if available)
+    #[cfg(since_api = "4.4")]
+    if info.godot_params.icon_path.is_null() {
+        if let Some(extension_icon) = crate::init::extension_config().icon {
+            let icon_gstring = crate::builtin::GString::from(extension_icon);
+
+            // SAFETY: The GString is stored in a static HashMap, so the pointer remains valid
+            // even after the lock guard is dropped. We must retrieve the pointer while holding
+            // the lock to ensure the HashMap isn't being modified concurrently.
```

Comment:
as above – either make it a note or mark unsafe thing – which requires respecting safety remarks to not make it unsound – with unsafe block